### PR TITLE
Feature/user input data

### PR DESCRIPTION
**Pull request by leighlars**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Styling- no new features
- [X] Refactoring
- [ ] Planning

### Detailed Description
I added HTML /CSS for the User to input their water consumption for the day on the info face of the card. I also experimented with adding methods in User to get the week's average of water consumption and each week day's consumption.
I added a method to HydroRepo and User Repo to instantiate all the data as an object of the appropriate class on load. 
I removed the glasses of water consumed each day (not part of spec) and replaced it with the input box. I also added a default msg to the calendar face that encouraged the User to drink more water if they did not have any water inputs for the week. 

### Why is this change required? What problem does it solve?
User can input their ounces of water for the day, and see their average water consumption over the 7 entries. 

### Were there any challenges that arose while implementing this feature? If so, how were the addressed?
Yes, it was very hard to figure out the logic of displaying the past week's ounces of water per day, and I could not figure it out. While the week's average can be displayed, each day of the week display is not working. This was my attempt to help Linus and does not need to be merged if not wanted. 

### Files modified:
- [ ] index.html
- [ ] .scss // file name(s):
- [ ] classes // file name(s):
- [X] test // files name(s): User, Hydration Hydration Repo, User Repo
- [X] scripts.js
- [ ] Other files:

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,6 +32,7 @@
     "webpack-dev-server": "^3.11.0"
   },
   "dependencies": {
-    "jquery": "^3.5.1"
+    "jquery": "^3.5.1",
+    "moment": "^2.27.0"
```
Comment (Leasw144) [inline package.json:36]: Is this that webpack you were talking about that gives the proper date?

File: src/User.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
+  constructor(userDetails) {
+    this.id = this.checkUserId(userDetails.id);
+    this.name = this.checkName(userDetails.name);
+    this.address = userDetails.address || 'No address added.';
+    this.email = userDetails.email || 'No email address added.';
+    this.strideLength = userDetails.strideLength || 'Stride length not added.';
+    this.dailyStepGoal = userDetails.dailyStepGoal || 'Daily step goal not added.';
+    this.friends = userDetails.friends || 'Add friends for friendly competition!';
     this.ouncesAverage = 0;
     this.ouncesRecord = [];
```
Comment (Leasw144) [inline src/User.js:15]: We talked a bit about renaming this so let's hit that on the next standup. Even recordOfOunces would be acceptable

File: src/css/styles.scss [hunk 3]
```diff
@@ -171,14 +193,25 @@ main {
...
   font-size: 2vw;
   text-align: center;
   width: 20vw;
 }
 
 /* BUTTONS */
-.info-button {
+.info-button,
+.friends-button,
+.calendar-button,
```
Comment (Leasw144) [inline src/css/styles.scss:211]: So with the SCSS lesson in mind, are we going to need to go back and refactor our file?

**Comment by Leasw144** [comment]
Great work on getting all of this working

File: src/HydrationRepository.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    this.hydrationData = this.instantiateHydroData(rawHydrationData);
+  }
+
+  instantiateHydroData(rawHydrationData) {
+    return rawHydrationData.map((data) => new Hydration(data));
+  }
+}
+
+
+export default HydrationRepository;
```
Comment (NakiNorton) [inline src/HydrationRepository.js:14]: Thanks for creating this new class. It'll make it easier to manage the raw data.

File: src/User.js (hunk 2 above)
Comment (NakiNorton) [reply src/User.js:15]: Yeah I agree we should rename this.

File: src/css/styles.scss [hunk 5]
```diff
@@ -156,6 +156,28 @@ main {
...
+  padding: 2px;
+  width: 75px;
+  height: 50px;
+}
+
+::placeholder {
+  color: white;
+  text-align: right;
+}
+
```
Comment (NakiNorton) [inline src/css/styles.scss:180]: This is looking great in the browser!

**Pull request merged**